client/layout/index.jsx

```diff
@@ -155,11 +156,19 @@ const Layout = createReactClass( {
 				'is-active': this.props.isLoading,
 			} );
 
+		const { recentSitesPreference, primarySiteId } = this.props;
+
+		let siteIdToFetch = get( recentSitesPreference, '0', null );
+
+		if ( ! siteIdToFetch ) {
+			siteIdToFetch = primarySiteId;
+		}
+
 		return (
 			<div className={ sectionClass }>
 				<DocumentHead />
 				<SitesListNotices />
-				<QuerySites allSites />
+				<QuerySites siteId={ siteIdToFetch } />
```

Comment:
I'd like to better understand why you query the site `recent[0] || primary` here. Is it because the "My Sites" masterbar link points to this site when you start at `/`, i.e., in Reader?

Would anything else break if you didn't query any site here?

Reply:
If you go to wp.com, you are presented with Reader. Normally, all sites are loaded here. If you immediately click on "My Sites" (which most of us do, I guess), you are taken to the all sites view because the all sites data haven't been loaded yet.

In this PR, we quickly fetch just a single site when you land on wp.com (Reader) so when you click on My Sites, you are navigated to that one site, not all sites view.

Now, what site to load? I assumed that it's either the recently selected one (that's why `recent[0]`) or, if `/me/preferences` has not been loaded yet (see `<QueryPreferences />` just one line below), we use the user's primary site ID. User object is surely loaded by that time as it's the first thing that gets fetched and Masterbar/layout wouldn't render without it.

For people with just one site, primary and recent will be the same. For us, it doesn't matter that much I think. You just want fast access to My Sites.